Mathlib/NumberTheory/EllipticDivisibilitySequence.lean

```diff
@@ -109,49 +114,129 @@ private def normEDS'' (b c d : R) : ℕ → R
     have h2 : m + 2 < n + 5 := (lt_add_one _).trans h3
     have h1 : m + 1 < n + 5 := (lt_add_one _).trans h2
     if hn : Even n then
-      normEDS'' b c d (m + 4) * normEDS'' b c d (m + 2) ^ 3 * (if Even m then b ^ 4 else 1) -
-        normEDS'' b c d (m + 1) * normEDS'' b c d (m + 3) ^ 3 * (if Even m then 1 else b ^ 4)
+      preNormEDS' b c d (m + 4) * preNormEDS' b c d (m + 2) ^ 3 * (if Even m then b else 1) -
+      preNormEDS' b c d (m + 1) * preNormEDS' b c d (m + 3) ^ 3 * (if Even m then 1 else b)
     else
       have h5 : m + 5 < n + 5 := add_lt_add_right
         (Nat.div_lt_self (Nat.odd_iff_not_even.mpr hn).pos <| Nat.lt_succ_self 1) 5
-      normEDS'' b c d (m + 2) ^ 2 * normEDS'' b c d (m + 3) * normEDS'' b c d (m + 5) -
-        normEDS'' b c d (m + 1) * normEDS'' b c d (m + 3) * normEDS'' b c d (m + 4) ^ 2
+      preNormEDS' b c d (m + 2) ^ 2 * preNormEDS' b c d (m + 3) * preNormEDS' b c d (m + 5) -
+      preNormEDS' b c d (m + 1) * preNormEDS' b c d (m + 3) * preNormEDS' b c d (m + 4) ^ 2
 
 variable (b c d : R)
 
+@[simp]
+lemma preNormEDS'_zero : preNormEDS' b c d 0 = 0 :=
+  rfl
+
+@[simp]
+lemma preNormEDS'_one : preNormEDS' b c d 1 = 1 :=
+  rfl
+
+@[simp]
+lemma preNormEDS'_two : preNormEDS' b c d 2 = 1 :=
+  rfl
+
+@[simp]
+lemma preNormEDS'_three : preNormEDS' b c d 3 = c :=
+  rfl
+
+@[simp]
+lemma preNormEDS'_four : preNormEDS' b c d 4 = d :=
+  rfl
+
+lemma preNormEDS'_odd (m : ℕ) : preNormEDS' b c d (2 * (m + 2) + 1) =
+    preNormEDS' b c d (m + 4) * preNormEDS' b c d (m + 2) ^ 3 * (if Even m then b else 1) -
+    preNormEDS' b c d (m + 1) * preNormEDS' b c d (m + 3) ^ 3 * (if Even m then 1 else b) := by
+  rw [show 2 * (m + 2) + 1 = 2 * m + 5 by rfl, preNormEDS', dif_pos <| even_two_mul _]
+  simp only [m.mul_div_cancel_left two_pos]
+
+lemma preNormEDS'_even (m : ℕ) : preNormEDS' b c d (2 * (m + 3)) =
+    preNormEDS' b c d (m + 2) ^ 2 * preNormEDS' b c d (m + 3) * preNormEDS' b c d (m + 5) -
+    preNormEDS' b c d (m + 1) * preNormEDS' b c d (m + 3) * preNormEDS' b c d (m + 4) ^ 2 := by
+  rw [show 2 * (m + 3) = 2 * m + 1 + 5 by rfl, preNormEDS', dif_neg m.not_even_two_mul_add_one]
+  simp only [Nat.mul_add_div two_pos]
+  rfl
+
+/-- The auxiliary sequence for a normalised EDS `W : ℤ → R`,
+with initial values `W(0) = 0`, `W(1) = 1`, `W(2) = 1`, `W(3) = c`, and `W(4) = d`.
+
+This extends `preNormEDS'` by defining its values at negative integers. -/
+def preNormEDS (n : ℤ) : R :=
+  n.sign * preNormEDS' b c d n.natAbs
```

Comment:
The `natAbs` is unnecessary here and I removed it in #13057. (I also corrected some capitalization like `IsEllSequence_id` there.)